src/main/java/myPackage/Deadlines.java

```diff
@@ -0,0 +1,56 @@
+package myPackage;
+
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task{
+    private String deadline;
+    private LocalDate localDate;
+
+    public Deadlines(String description, String deadline) {
+        super(description);
+        try {
+            String[] str = deadline.split("by ");
+            localDate = LocalDate.parse(str[1]);
+            this.deadline = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        }
+        catch (Exception e) {
+            System.out.println("not proper date");
+            this.deadline = deadline;
+        }
+        System.out.printf("[D][ ] %s (%s)", this.description, this.deadline);
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
```

Comment:
Is `this` keyword redundant here?
```suggestion
        isDone = true;
```